tests/unittests/Foundation/NSNotificationQueueTests.mm

```diff
@@ -0,0 +1,206 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Foundation/Foundation.h>
+#import <TestFramework.h>
+#import <Starboard/SmartTypes.h>
+
+@interface NSNotificationQueueTestHelper : NSObject {
+    NSRunLoop* _loop;
+    BOOL _isStopped;
+    NSCondition* _threadStartedCondition;
+}
+@property (retain) NSThread* thread;
+- (void)scheduleAndAwaitBlock:(void (^)())block;
+@end
+
+@implementation NSNotificationQueueTestHelper
+
+- (instancetype)init {
+    if (self = [super init]) {
+        _thread = [[NSThread alloc] initWithTarget:self selector:@selector(spinRunLoop) object:nil];
+        _threadStartedCondition = [[NSCondition alloc] init];
+        [_threadStartedCondition lock];
+        [_thread start];
+        [_threadStartedCondition wait];
```

Comment:
possibly just move my _NSBooleanCondition helper out of NSOperationTests (I still wish we had NSConditionLock...)